src/facilities/costCenters.spec.js

```diff
@@ -0,0 +1,540 @@
+          `${expectedHost}/costcenters`,
+          formattedCostCenterToServer
+        );
+      });
+
+      it('formats the returned cost center object', function() {
+        return promise.then(() => {
+          expect(formatCostCenterFromServer).to.be.calledWith(
+            formattedCostCenterFromServer
+          );
+        });
+      });
+
+      it('returns a fulfilled promise with the new cost center information', function() {
+        return expect(promise).to.be.fulfilled.and.to.eventually.equal(
+          expectedCostCenter
+        );
+      });
+    });
+
+    context('when there is missing required information', function() {
+      ['name', 'organizationId'].forEach(function(field) {
+        it(`it throws an error when ${field} is missing`, function() {
+          const costCenter = fixture.build('costCenter');
+          const costCenters = new CostCenters(
+            baseSdk,
+            baseRequest,
+            expectedHost
+          );
+          const promise = costCenters.create(omit(costCenter, [field]));
+
+          return expect(promise).to.be.rejectedWith(
+            `A ${field} is required to create a new cost center.`
+          );
+        });
+      });
+    });
+  });
+
+  describe('delete', function() {
+    context('when all required information is supplied', function() {
+      let expectedCostCenterId;
+      let promise;
+
+      beforeEach(function() {
+        expectedCostCenterId = fixture.build('costCenter').id;
+
+        const costCenters = new CostCenters(baseSdk, baseRequest, expectedHost);
+        promise = costCenters.delete(expectedCostCenterId);
+      });
+
+      it('requests to delete the cost center', function() {
+        expect(baseRequest.delete).to.be.calledWith(
+          `${expectedHost}/costcenters/${expectedCostCenterId}`
+        );
+      });
+
+      it('returns a fulfilled promise', function() {
+        return expect(promise).to.be.fulfilled;
+      });
+    });
+
+    context('when there is missing required information', function() {
+      it(`it throws an error when the cost center id is missing`, function() {
+        const expectedErrorMessage = `A cost center id is required for deleting a cost center.`;
+
+        const costCenters = new CostCenters(baseSdk, baseRequest, expectedHost);
+        const promise = costCenters.delete();
+
+        return expect(promise).to.be.rejectedWith(expectedErrorMessage);
+      });
+    });
+  });
+
+  describe('getAll', function() {
+    let expectedCostCenter;
+    let formatCostCenterFromServer;
+    let costCentersFromServer;
+    let promise;
+    let request;
+
+    beforeEach(function() {
+      const numberOfCostCenters = faker.random.number({
+        min: 1,
+        max: 10
+      });
+      expectedCostCenter = fixture.buildList('costCenter', numberOfCostCenters);
+      costCentersFromServer = fixture.buildList(
+        'costCenter',
+        numberOfCostCenters
+      );
```

Comment:
Should do `costCentersFromServer = fixture.buildList('costCenter', numberOfCostCenters, null, { fromServer: true });` (I think that will work, but my brain is still a little out of it from being off work)